source-postgres/datatype_test.go

```diff
@@ -68,7 +68,8 @@ func TestDatatypes(t *testing.T) {
 		// {ColumnType: `timestamp with time zone`, ExpectType: `{"type":"string","format":"date-time"}`, InputValue: `'January 8, 1999'`, ExpectValue: `"1999-01-07T16:00:00-08:00"`},
 		{ColumnType: `time`, ExpectType: `{"type":["integer","null"]}`, InputValue: `'04:05:06 PST'`, ExpectValue: `14706000000`},
```

Comment:
@willdonnelly these test cases are odd -- this is outputting a unix timestamp? Why that, as opposed to `04:05:06--8` (and also, we don't know the date, right) ?

Reply:
@jgraettinger I think this is because the `ExpectType` is set to `integer`

Reply:
yep, I'm asking _why_ are we mapping these into an integer? Seems they should be RFC3339, no?

Reply:
@jgraettinger ah sorry. So in this case, the column type is `time`, which is time without timezone.

pgx handles this value as a unix timestamp: https://pkg.go.dev/github.com/jackc/pgtype#Time

Since it doesn't have timezone, to format it as RFC3339 would require us to assign it an arbitrary timezone (UTC being the natural choice), but that is not really accurate, since the time does not really have a timestamp.

Reply:
> pgx handles this value as a unix timestamp: https://pkg.go.dev/github.com/jackc/pgtype#Time

Ah -- and to pedantically clarify it's not a unix timestamp but is instead the number of microseconds since midnight. Very different things, but I do follow that that implies a representation like "04:05:06.123456", which also would need to omit a timezone offset (because the timezone is unknown), and is thus not a valid `format: "time"`, and in that sense just spitting out the number of microseconds instead is reasonable.

For other DB's we asses the system's timezone or allow it to be configured and use that to resolve the ambiguity, but this was definitely written before we established that practice. Thanks

Reply:
Yeah, this logic predates all of the work on supporting configurable time zones for this sort of timezone-free datatype.

If this were being implemented for the first time today I'd probably favor converting the microseconds count into an RFC3339 time with appropriate time zone conversion, but that's probably beyond the scope of this PR to do now since it would require making sure there aren't any existing uses that would be broken.